infra/presubmit.py

```diff
@@ -323,14 +323,29 @@ def yapf(paths, validate=True):
 
 def get_changed_files():
   """Return a list of absolute paths of files changed in this git branch."""
-  # FIXME: This doesn't work if branch is behind master.
-  diff_command = ['git', 'diff', '--name-only', 'FETCH_HEAD']
-  return [
-      os.path.abspath(path)
-      for path in subprocess.check_output(diff_command).decode().splitlines()
-      if os.path.isfile(path)
+  main_branch = subprocess.check_output(
+      ['git', 'rev-parse', '--abbrev-ref', 'origin/HEAD']).strip().decode()
+  branch_commit_hash = subprocess.check_output(
+      ['git', 'merge-base', 'HEAD', main_branch]).strip().decode()
+
+  diff_commands = [
+      # Return list of modified files in the commits on this branch.
+      ['git', 'diff', '--name-only', branch_commit_hash + '..'],
+      # Return list of modified files from uncommitted changes.
```

Comment:
The uncommitted part strikes me as unnecessary, even though it may help with local debugging is it worth the complexity?

Reply:
yes we will keep forgetting about this, so adding the exact command. this happens often enough?

Reply:
>this happens often enough?

It does?

Reply:
Lets say you upload a PR.
When you want to fix something else in it, you make a change, then run presubmit, and then create local commit. If i remove it, it would mean create change, commit locally, run presubmit, then git commit -a --amend to amend, so this step addition is frustating.

Reply:
Oh, I got confused. I didn't realize we were discussing presubmit. I thought we were discussing the build images scriptwe use in CI.
I agree 100% about your point in this file. presubmit needs to work on uncommitted changed